entity-framework/core/modeling/relationships/many-to-many.md

```diff
@@ -0,0 +1,1071 @@
+
+This `PostTag.Id` property is now picked up as the primary key by convention, so the only configuration needed is a call to `UsingEntity` for the `PostTag` type:
+
+<!--
+            protected override void OnModelCreating(ModelBuilder modelBuilder)
+            {
+                modelBuilder.Entity<Post>()
+                    .HasMany(e => e.Tags)
+                    .WithMany(e => e.Posts)
+                    .UsingEntity<PostTag>();
+            }
+-->
+[!code-csharp[ManyToManyWithJoinClassHavingPrimaryKeyConfig](../../../../samples/core/Modeling/Relationships/ManyToMany.cs?name=ManyToManyWithJoinClassHavingPrimaryKeyConfig)]
+
+And the resulting schema for the join table is:
+
+```sql
+CREATE TABLE "PostTag" (
+    "Id" INTEGER NOT NULL CONSTRAINT "PK_PostTag" PRIMARY KEY AUTOINCREMENT,
+    "PostId" INTEGER NOT NULL,
+    "TagId" INTEGER NOT NULL,
+    CONSTRAINT "FK_PostTag_Posts_PostId" FOREIGN KEY ("PostId") REFERENCES "Posts" ("Id") ON DELETE CASCADE,
+    CONSTRAINT "FK_PostTag_Tags_TagId" FOREIGN KEY ("TagId") REFERENCES "Tags" ("Id") ON DELETE CASCADE);
+```
+
+A primary key can also be added to the join entity without defining a class for it. For example, with just `Post` and `Tag` types:
+
+<!--
+        public class Post
+        {
+            public int Id { get; set; }
+            public List<Tag> Tags { get; } = new();
+        }
+
+        public class Tag
+        {
+            public int Id { get; set; }
+            public List<Post> Posts { get; } = new();
+        }
+-->
+[!code-csharp[ManyToManyWithPrimaryKeyInJoinEntity](../../../../samples/core/Modeling/Relationships/ManyToMany.cs?name=ManyToManyWithPrimaryKeyInJoinEntity)]
+
+The key can be added with this configuration:
+
+<!--
+            protected override void OnModelCreating(ModelBuilder modelBuilder)
+            {
+                modelBuilder.Entity<Post>()
+                    .HasMany(e => e.Tags)
+                    .WithMany(e => e.Posts)
+                    .UsingEntity(
+                        j =>
+                        {
+                            j.IndexerProperty<int>("Id");
+                            j.HasKey("Id");
+                        });
+            }
+-->
+[!code-csharp[ManyToManyWithPrimaryKeyInJoinEntityConfig](../../../../samples/core/Modeling/Relationships/ManyToMany.cs?name=ManyToManyWithPrimaryKeyInJoinEntityConfig)]
+
+Which results in a join table with a separate primary key column:
+
+```sql
+CREATE TABLE "PostTag" (
+    "Id" INTEGER NOT NULL CONSTRAINT "PK_PostTag" PRIMARY KEY AUTOINCREMENT,
+    "PostsId" INTEGER NOT NULL,
+    "TagsId" INTEGER NOT NULL,
+    CONSTRAINT "FK_PostTag_Posts_PostsId" FOREIGN KEY ("PostsId") REFERENCES "Posts" ("Id") ON DELETE CASCADE,
+    CONSTRAINT "FK_PostTag_Tags_TagsId" FOREIGN KEY ("TagsId") REFERENCES "Tags" ("Id") ON DELETE CASCADE);
+```
+
+### Many-to-many and join table with payload
```

Comment:
This feels important/common, consider putting it much higher up (some of the above scenarios feel very niche)